release/script_release.py

```diff
@@ -71,6 +71,7 @@ def __init__(self, release_type, remote_name="canalTP"):
         self.git.rebase(remote_name + "/dev", "dev")
         self.dev_data_version = self.get_data_version()
         try:
+            self.git.pull("release")
```

Comment:
1. In case someone have done extra unpushed work on `release` branch locally (and forgot about it) : this should be `--ff-only` as one would push weird things if a merge commit is needed between local `release` and `CanalTP:release`.

2. I guess the rebase is not needed anymore. If it still is, the same may be done there.

3. We should precise the distant release branch we want to pull from: in case referent distant repo is personal one and not `remote_name` (CanalTP).

Reply:
I've removed `git.pull("release")` in favour of `git checkout -B <local branch> <remote branch>`

This has 2 benefits:

1. It creates a local branch if it doesn't exist
2. it reset the local branch with the remote one

https://git-scm.com/docs/git-checkout#Documentation/git-checkout.txt-emgitcheckoutem-b-Bltnewbranchgtltstartpointgt